### PR TITLE
Add Link Command and Unlink Command

### PR DESCRIPTION
**Pull request by LiuFangrui**
## Add Link Command
Link Command will link Person at index `PERSON_INDEX` and Internship at `INTERNSHIP_INDEX` together
- Command checks for errors such as person and internship provided are already linked to themselves/other objects
- Add test cases for Link Command

## Add Unlink Command
Unlink Command will unlink Person at index `PERSON_INDEX` and Internship at `INTERNSHIP_INDEX`
- Users are allowed to pass in 1 `PERSON_INDEX` or `INTERNSHIP_INDEX` using e.g `unlink p/1` or both `PERSON_INDEX` and `INTERNSHIP_INDEX` using e.g `unlink p/1 i/1`
- Command checks for errors such as person and/or internship provided are not linked to each other
- Add test cases for Unlink Command

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/LinkCommand.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    public static final String COMMAND_WORD = "link";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Links a Person and an Internship. "
+            + "Parameters: "
+            + PREFIX_PERSON + "PERSON_INDEX "
+            + PREFIX_INTERNSHIP + "INTERNSHIP_ID "
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_PERSON + "0 "
+            + PREFIX_INTERNSHIP + "0 ";
+
```
Comment (chengda300) [inline src/main/java/seedu/address/logic/commands/LinkCommand.java:32]: CI claimed that there is trailing whitespace here, might want to remove it so it passes

Comment (chengda300) [inline src/main/java/seedu/address/logic/commands/LinkCommand.java:32]:
```suggestion

```
CI says there are 4 spaces here so it failed

**Comment by chengda300** [comment]
Generally it looks correct, I trust you
Though if needed, internshipID / personID might be changed to use the InternshipId(null) / PersonId(null) possibly in v1.4 if there are related null detection bugs.

**Review by chengda300 — APPROVED** [review]

**Comment by chengda300** [comment]
Fix merge conflict due to import statement changes

**Comment by nqt230** [comment]
LGTM 👍

**Pull request merged**